### PR TITLE
feat: Don't wrap the original Promise

### PR DESCRIPTION
**Pull request by dubzzz**
PR opened following a discussion with @sheremet-va on https://github.com/tinylibs/tinyspy/pull/44. I squashed all my commits together before opening the PR.

### CODE REVIEW

File: test/index.test.ts [hunk 1]
```diff
@@ -461,12 +464,47 @@ test('async error', async () => {
...
-  expect(spy.results[0][0]).toEqual('error')
-  expect(spy.results[0][1].message).toEqual('async error')
+  expect(spy.results).toEqual([['ok', promise]])
+  expect(spy.resolves[0][0]).toEqual('error')
+  expect(spy.resolves[0][1].message).toEqual('async error')
   expect(caughtError).toBeInstanceOf(Error)
   expect(caughtError?.message).toEqual('async error')
 })
 
+test.only('async order preserved', async () => {
```
Comment (sheremet-va) [inline test/index.test.ts:474]: `.only` are not allowed, not sure why CI doesn't fail

**Review by sheremet-va — CHANGES_REQUESTED** [review]
Do you also want to add a readme? Or should I?

File: test/index.test.ts (hunk 1 above)
Comment (dubzzz) [reply test/index.test.ts:474]: Woups. I missed it

Comment (dubzzz) [inline test/index.test.ts:474]:
```suggestion
test('async order preserved', async () => {
```

**Comment by dubzzz** [comment]
> Do you also want to add a readme? Or should I?

I can but I'll have a limited availability until Monday included

**Comment by dubzzz** [comment]
And I'll need pointers

**Pull request merged**